RealmSwift/Query.swift

```diff
@@ -289,6 +289,23 @@ extension Query where T == Bool {
     }
 }
 
+// MARK: Mixed
+
+extension Query where T == AnyRealmValue {
+    /// :nodoc:
+    public subscript(position: Int) -> Query<AnyRealmValue> {
+        .init(appendKeyPath("[\(position)]", options: [.isPath]))
+    }
+    /// :nodoc:
+    public subscript(key: String) -> Query<AnyRealmValue> {
+        .init(appendKeyPath("['\(key)']", options: [.isPath]))
```

Comment:
This needs to handle `key` containing quotes.

Reply:
Do you mean because of this 
```
Note
The expressions you write inside parentheses within an interpolated string can’t contain an unescaped backslash (\), a carriage return, or a line feed. However, they can contain other string literals.
```

https://docs.swift.org/swift-book/documentation/the-swift-programming-language/stringsandcharacters/

Reply:
If you do `$0.anyCol["'"] == 0` to query on the dictionary key "'" this will produce invalid syntax.

Reply:
`$0.anyCol["'"]` is not allowed for AnyRealmValue subscripts

Reply:
I don't know what you mean. `any["'"] == %@` (or `any['\''] == %@`) is a perfectly valid query, which this currently fails to generate a valid string for.